consensus/state.go

```diff
@@ -2308,22 +2308,23 @@ func (cs *State) signVote(
 		BlockID:          types.BlockID{Hash: hash, PartSetHeader: header},
 	}
 
-	if msgType == tmproto.PrecommitType && len(vote.BlockID.Hash) != 0 {
+	extEnabled := cs.state.ConsensusParams.ABCI.VoteExtensionsEnabled(cs.Height)
```

Comment:
move this variable to line 2109 and simplify 2111.

Reply:
I can't really move the variable there (another function), but I'll copy its definition